src/librustc_mir/dataflow/impls/borrowed_locals.rs

```diff
@@ -93,19 +93,15 @@ struct BorrowedLocalsVisitor<'gk> {
 }
 
 fn find_local(place: &Place<'_>) -> Option<Local> {
-    place.iterate(|place_base, place_projection| {
-        for proj in place_projection {
-            if proj.elem == ProjectionElem::Deref {
-                return None;
-            }
-        }
+    if place.is_indirect() {
+        return None;
+    }
 
-        if let PlaceBase::Local(local) = place_base {
-            Some(*local)
-        } else {
-            None
-        }
-    })
+    if let PlaceBase::Local(local) = place.base {
+        Some(local)
+    } else {
+        None
+    }
```

Comment:
```suggestion
        match place.base {
        PlaceBase::Local(local) if !place.is_indirect() => Some(local),
        _ => None,
    }
```
A more streamlined version of this...